src/dns_parser.rs

```diff
@@ -771,14 +771,18 @@ impl DnsOutgoing {
     //    server/responder SHOULD include the following additional records:
 
     //    o  All address records (type "A" and "AAAA") named in the SRV rdata.
-    pub(crate) fn add_additional_answer(&mut self, answer: DnsRecordBox) {
+    pub(crate) fn add_additional_answer(&mut self, answer: impl DnsRecordExt + 'static) {
```

Comment:
Although I'm not sure about the `static dispatch` thing, I see the benefit of not requiring the callers to add `Box::new()` at the call site.

And I think I prefer to have `Send` in here, compared to move `Send` into `DnsRecordExt` trait, as I mentioned in another comment.